File: CourseManagementSystem.API/ClientApp/src/app/viewmodels/courseMemberVM.ts
export class PersonIdVM {
  /**
   * identifier of the person
   */
  public id: string;
}

/**
 * viewmodel representing member of a course
 */
export class CourseMemberVM extends PersonIdVM {
  /**
   * name of the person
   */
  public name: string;

  /**
   * user's email
   */
  public email: string;
}

File: CourseManagementSystem.API/ViewModels/CourseMemberVM.cs
using CourseManagementSystem.API.Validation.Attributes;
using System.ComponentModel.DataAnnotations;

namespace CourseManagementSystem.API.ViewModels
{
    /// <summary>
    /// viewmodel representing member of a course
    /// </summary>
    public class CourseMemberVM
    {
        public CourseMemberVM()
        {
        }

        public CourseMemberVM(string id, string name, string email)
        {
            Id = id;
            Name = name;
            Email = email;
        }

        /// <summary>
        /// identifier of the person
        /// </summary>
        [RequiredWithDefaultErrorMessage]
        public string Id { get; set; }

        /// <summary>
        /// name of the person
        /// </summary>
        [RequiredWithDefaultErrorMessage]
        public string Name { get; set; }

        /// <summary>
        /// email of the person
        /// </summary>
        [RequiredWithDefaultErrorMessage]
        [DataType(DataType.EmailAddress)]
        public string Email { get; set; }
    }
}